src/main/java/jewellery/inventory/service/UserInOrganizationService.java

```diff
@@ -0,0 +1,198 @@
+      throw new MissingOrganizationPermissionException(
+          currentUser.getId(), organization.getId(), permission);
+    }
+    logger.debug(
+        "User permission validation successful. User ID: {}, Organization ID: {}",
+        currentUser.getId(),
+        organization.getId());
+  }
+
+  private void validateUserInOrganization(Organization organization) {
+    User currentUser = userService.getUser(authService.getCurrentUser().getId());
+    boolean isUserInOrganization =
+        organization.getUsersInOrganization().stream()
+            .anyMatch(userInOrganization -> userInOrganization.getUser().equals(currentUser));
+
+    if (!isUserInOrganization) {
+      throw new UserIsNotPartOfOrganizationException(currentUser.getId(), organization.getId());
+    }
+    logger.debug(
+        "User permission validation successful. User ID: {}, Organization ID: {}",
+        currentUser.getId(),
+        organization.getId());
+  }
+
+  private boolean hasPermission(
+      User user, Organization organization, OrganizationPermission permission) {
+    return organization.getUsersInOrganization().stream()
+        .anyMatch(
+            userInOrganization ->
+                userInOrganization.getUser().equals(user)
+                    && userInOrganization.getOrganizationPermission().contains(permission));
+  }
+
+  private UserInOrganization createUserInOrganization(
+      UserInOrganizationRequestDto requestDto, Organization organization) {
+    UserInOrganization userInOrganization = new UserInOrganization();
+    userInOrganization.setUser(userService.getUser(requestDto.getUserId()));
+    userInOrganization.setOrganization(organization);
+    userInOrganization.setOrganizationPermission(requestDto.getOrganizationPermission());
+    userInOrganizationRepository.save(userInOrganization);
+    logger.debug(
+        "Successfully created user in organization. User ID: {}, Organization ID: {}, Organization Permission: {}",
+        userInOrganization.getUser().getId(),
+        userInOrganization.getOrganization().getId(),
+        userInOrganization.getOrganizationPermission());
+    return userInOrganization;
+  }
+
+  private void addUserToOrganization(
+      UserInOrganization userInOrganization, Organization organization) {
+    organization.getUsersInOrganization().add(userInOrganization);
+    logger.info(
+        "Successfully added user in the organization. Organization ID: {}, User ID: {}",
+        organization.getId(),
+        userInOrganization.getUser().getId());
+    organizationService.saveOrganization(organization);
+  }
+
+  private void changeUserPermissionInOrganization(
+      Organization organization,
+      User userForUpdate,
+      List<OrganizationPermission> organizationPermissionList) {
+
+    organization.getUsersInOrganization().stream()
+        .filter(userInOrg -> userInOrg.getUser().equals(userForUpdate))
+        .findFirst()
+        .ifPresent(
```

Comment:
Are we certain here we are finding a user here every time? If we don't find a user this method will just pass without any errors. Should we try to protect ourselves by throwing and exception if it's not present?